proto/cosmos/app/v1alpha1/query.proto

```diff
@@ -3,12 +3,20 @@ syntax = "proto3";
 package cosmos.app.v1alpha1;
 
 import "cosmos/app/v1alpha1/config.proto";
+import "google/protobuf/descriptor.proto";
+import "cosmos/query/v1/query.proto";
 
 // Query is the app module query service.
 service Query {
 
   // Config returns the current app config.
   rpc Config(QueryConfigRequest) returns (QueryConfigResponse) {}
+
+  // FileDescriptorSet returns the full FileDescriptorSet of the app in order
+  // to enable easier generation of dynamic clients.
+  rpc FileDescriptorSet(QueryFileDescriptorSetRequest) returns (QueryFileDescriptorSetResponse) {
```

Comment:
why not add this to https://github.com/cosmos/cosmos-sdk/blob/75475a4f7261203d40a10571602d59cfbfe04d2f/proto/cosmos/base/reflection/v2alpha1/reflection.proto#L7? seems like they go together

Reply:
Or reflection.v1beta1.

I would put cosmos-related stuff here (e.g. chain id), and grpc-related stuff (e.g. file descriptors) in cosmos.base.reflection

Reply:
agree here. I would like this approach as it cleans up some random grpc services

Reply:
I like this approach

Reply:
moved to `cosmos.reflection.v1`. how does that look?